Some stuff for the web interface - attempt 2

hhm0 opened the pull request:
Same as pull https://github.com/MalcolmRobb/dump1090/pull/17 but from another branch in local repo besides master; learning the Github ropes :-).

hhm0 committed:
web interface

- more configurable parameters
	- plane ttls
	- OSM tiles source URL
	- AJAX data source URL
- remember more ui settings in local storage
	- which map type was viewed
	- plane table sorting direction, field
- ui changes
	- indicates when plane on ground
	- shows empty box in table when not receiving value (like CLI interactive UI)
- fixing etc.
	- plane was selected, ttl'ed, then came back, had selected marker color
	- memory leaking (orphan dom nodes, seemingly), so remove old elements on refresh
	- more robust handling for server disconnects, restarts
- code refactoring
	- refactor dynamically generated HTML elements and create DOM elements instead
	- added functions for new code used multiple times

hhm0 committed:
prefix instead of format, always (re)render planes url

- prefix easier to comprehend than format for URL
- AJAX url can change in between so always rerender

hhm0 committed:
B"H handle missing JSON plane fields better

hhm0 committed:
B"H use separate configs per URL

hhm0 committed:
Merge branch 'master' into mods

mindlesstux commented:
Looks readable and a nice improvement to my typos and code.

The only thing I am confused by is the OSM code changes.  I may just not be seeing the meaning behind the changes themselves.

Is there a reason you took the URL into its own variable?  I just used the example code from http://wiki.openstreetmap.org/wiki/Google_Maps_Example to get that layer working.

Otherwise I think it looks good, but wont be able test it out till at least this weekend.  Granted I tend to favor my new reworking of the page (slowly in progress) as the table of planes was kind of a hack if I remember right and bogs down with a lot of planes on the map.

mindlesstux commented on public_html/extension.js:
```diff
@@ -1,7 +1,7 @@
 //
 // This file is so users can modify how the page acts
-// without diving to deep in the code.  This way we can
+// without diving too deep in the code.  This way we can
```
Nit pick,
"without diving to deep in the code."
vrs yours
"without diving also deep in the code."
(unless I am forgetting some crazy english rule with 'too', and yes I know I make a lot of mispelling/typos.)

mindlesstux commented on public_html/planeObject.js:
```diff
@@ -214,7 +249,7 @@ var planeObject = {
 					map: GoogleMap,
 					icon: this.funcGetIcon(),
-					visable: true
+					visible: true
```
I really got to kick myself in the ass for this one.  Good catch!
https://developers.google.com/maps/documentation/javascript/reference?csw=1#Data.StyleOptions

hhm0 replied:
@bdavenport,

Thanks for the responses!

The OSM config is for custom OSM tilesets, since there are many
different tile renderings (see
http://wiki.openstreetmap.org/wiki/Rendering for more info on this).
For example running "OsmTileUrlFormat =
'http://tile.stamen.com/toner-lite/{z}/{x}/{y}.png'" in the dev
console, changes the map immediately to use the toner-lite tileset
from maps.stamen.com.
The lastOsmX code is so the tilesource may be changed live, now from
browser developer console, and later thru the settings UI.

Also there is a more readable (without those indecipherable letter
variables and params) and more robust (working OK with old dump1090's
without many fields) version of the updVal code, in the "mods" branch
of my repo.

About the table, personally I like it, but if it got replaced with
something better than that's good :-) To speed it up (it is still slow
with many rows after applying the pull request), I was thinking of
reusing generated DOM nodes, and updating the rows themselves (keeping
a reference to a plane's row in each plane-object until ttl expired)
instead of generating the rows anew on every refresh.

On 6/2/14, Brian Davenport notifications@github.com wrote:

> > @@ -214,7 +249,7 @@ var planeObject = {
> >                      position: new google.maps.LatLng(this.latitude, this.longitude),
> >                      map: GoogleMap,
> >                      icon: this.funcGetIcon(),
> > -                    visable: true
> > -                    visible: true
> 
> I really got to kick myself in the ass for this one.  Good catch!
> https://developers.google.com/maps/documentation/javascript/reference?csw=1#Data.StyleOptions
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/MalcolmRobb/dump1090/pull/18/files#r13309038

mindlesstux commented:
See notes on,
https://github.com/MalcolmRobb/dump1090/pull/8#issuecomment-44916303

That makes sense on the OSM layer now.

Yeah I am trying to replace the table with something else that does searching column re-arranging, etc.  The couple of data grid/table plugins I have tried were not really up to what I was looking for.  The one on the adsb... page I have hope for but I have not done much with it in the past couple of months as I started a new job and any free time is spent elsewhere.

hhm0 commented:
On 6/2/14, Brian Davenport notifications@github.com wrote:

> See notes on,
> https://github.com/MalcolmRobb/dump1090/pull/8#issuecomment-44916303
> 
> That makes sense on the OSM layer now.
> 
> Yeah I am trying to replace the table with something else that does
> searching column re-arranging, etc.

Yes, I was thinking of column sorting (and also was thinking of drag n
drop), and also of extending the extension mechanism to make it
possible to add custom columns.

> The couple of data grid/table plugins I
> have tried were not really up to what I was looking for.  The one on the
> adsb... page I have hope for but I have not done much with it in the past
> couple of months as I started a new job and any free time is spent
> elsewhere.

By the way, what is the license for dump1090-helper (which it seems
runs adsb.mindlesstux.com)? Perhaps I can integrate some of the ideas
from it, if it is ok with you, if it is the same or compatible license
to dump1090 (https://github.com/MalcolmRobb/dump1090#credits).

P.S. The Ajax... URL configs were inspired by how dump1090-helper is a
frontend to data :-)

> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/MalcolmRobb/dump1090/pull/18#issuecomment-44916747

hhm0 commented:
On 6/3/14, h hm heehooman@gmail.com wrote:

> On 6/2/14, Brian Davenport notifications@github.com wrote:
> 
> > See notes on,
> > https://github.com/MalcolmRobb/dump1090/pull/8#issuecomment-44916303
> > 
> > That makes sense on the OSM layer now.
> > 
> > Yeah I am trying to replace the table with something else that does
> > searching column re-arranging, etc.
> 
> Yes, I was thinking of column sorting (and also was thinking of drag n
> drop), and also of extending the extension mechanism to make it
> possible to add custom columns.
> 
> > The couple of data grid/table plugins I
> > have tried were not really up to what I was looking for.  The one on the
> > adsb... page I have hope for but I have not done much with it in the past
> > couple of months as I started a new job and any free time is spent
> > elsewhere.
> 
> By the way, what is the license for dump1090-helper (which it seems
> runs adsb.mindlesstux.com)? Perhaps I can integrate some of the ideas
> from it, if it is ok with you, if it is the same or compatible license
> to dump1090 (https://github.com/MalcolmRobb/dump1090#credits).

Concerning the license, I could just use your dump1090
mapRefactorSettings branch and glean insights from there, since it is
under the license. It seems the web-ui there is not very different
from the one in -helper.

So the license is solved :-).

> P.S. The Ajax... URL configs were inspired by how dump1090-helper is a
> frontend to data :-)
> 
> > ---
> > 
> > Reply to this email directly or view it on GitHub:
> > https://github.com/MalcolmRobb/dump1090/pull/18#issuecomment-44916747

hhm0 committed:
Merge branch 'master' of https://www.github.com/MalcolmRobb/dump1090

hhm0 committed:
Merge branch 'master' into mods

hhm0 committed:
Merge remote-tracking branch 'origin/master' into web_update

hhm0 committed:
Merge remote-tracking branch 'origin/master' into mods

hhm0 committed:
Merge branch 'mods' into web_update

hhm0 committed:
B"H web ui: bugfix/cleanup - removing stale planes

hhm0 committed:
B"H web ui: more stale stuff

hhm0 committed:
B"H web ui: fix bug in plane updating

The "valid*" plane values should be preserved if had been set.

Setting this as a boolean even when set to a non-boolean value by user
removes values with special meaning, for example
validaltitude == -1 (on-ground).

hhm0 committed:
B"H web ui: fix altitude validity detection

hhm0 committed:
B"H web ui: recode handling of missing data

hhm0 committed:
B"H web ui: make plane update code easier to read

hhm0 committed:
Merge branch 'master' of https://www.github.com/MalcolmRobb/dump1090 into web_update

hhm0 committed:
B"H web ui: openstreetmap - use https tile source by default

hhm0 committed:
B"H web ui: backend - check ground validity before altitude validity

Before, even when an aircraft was on the ground, validaltitude was 1
instead of -1. This is because when MODES_ACFLAGS_ALTITUDE_VALID flag
was true, no further checking would be done, and validaltitude would be
set to 1. However, the MODES_ACFLAGS_ALTITUDE_VALID can be true when the
aircraft is on ground, so the ground status must be checked before the
altitude validity flag.

hhm0 commented:
Several fine-tunings have been made to this pull request. These fix small bugs in the older pull commits, and in general add polish to the pull code.

Note that this branch, despite its name `web_update`, is not intended to change the user interface, but to refine and enhance the UI and code. New features, and user-facing changes, have been minimal in this pull.

MalcolmRobb commented:
Guys,
The reason I've been tardy in merging these changes is that whilst I'm happy to incorporate changes made to the web/json/java interface, I'm nervose about the changes made to net_io.c.

You seem to be filtering out various binary bit fields from a->bFlags, and then snprintf-ing them to a text string to pass them out. The problem is that every time you want a different flag the net_io.c code needs changing with a new snprintf incorporating the new bits. 

Is it not possible to just pass the a->bFlags in it's entirity to your json, and then filter out the bits you want in there? That would allow the c code to remain unaltered from now on?
Cheers
Malcolm

hhm0 committed:
Merge branch 'master' of https://www.github.com/MalcolmRobb/dump1090 into web_update

hhm0 committed:
Merge branch 'master' of https://www.github.com/MalcolmRobb/dump1090 into web_update

ddeitterick committed:
Fix W3C Validation Errors

Fixed some W3C validation errors.

hhm0 committed:
B"H undo changes to net_io.c

The changes here to net_io.c are not critical, since part of the web
interface updating was to make it more robust and resilient to
missing/absent info.

However, the functionality added to net_io.c is still useful, since more
information can be deduced (for example, whether the plane is on the
ground).

It might be useful to separately commit/push the changes to net_io,
since in any case the modifications are useful for other things besides
the web ui.

hhm0 commented:
I pushed a commit which undoes changes to `net_io.c` -- see the commit msg (25be9bfc) for more info.

hhm0 committed:
B"H: public_html: integrate changes from ddeitterick at github

Changes for w3c validation testing.

hhm0 commented:
...and integrated the improvements at #34 into this pull.

hhm0 committed:
Merge remote-tracking branch 'ddeitterick/fix_w3c_validation' into web_update

hhm0 committed:
B"H Merge remote-tracking branch 'hhm/web_update' into web_update

Conflicts:
	public_html/gmap.html

hhm0 committed:
B"H Merge branch 'master' of https://www.github.com/MalcolmRobb/dump1090 into web_update

Conflicts:
	public_html/gmap.html
	public_html/script.js

hhm0 committed:
B"H adapt degree character display code to escaped text

hhm0 committed:
Merge branch 'master' of https://www.github.com/MalcolmRobb/dump1090 into web_update

MalcolmRobb commented:
I'm going to close this pull, and look at #37.

hhm0 commented:
This branch is still applicable, as it still has part of the original
pull request code.

The difference in this pull from how it worked before is that I
modified this pull to only change the web files, and broke the C code
out into #37.

If you go to the "files changed" for this pull(#18), you will see that
none of the C files are changed anymore.

I probably should have done this differently, by splitting the code
into _2_ new pulls (1. C json api code 2. html/js/css web pages) and
closing this. Instead I split the C code into #37 and _modified_ this
pull away from its original code so it only patches the web stuff.

On 10/28/14, MalcolmRobb notifications@github.com wrote:

> I'm going to close this pull, and look at #37.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/MalcolmRobb/dump1090/pull/18#issuecomment-60750624

MalcolmRobb commented:
The problem I have is the nature modification to the C code. Statements like ...
  if (a->bFlags & MODES_ACFLAGS_SPEED_VALID) { 
      speed = 1; 
  }

followed later on by a snprintf( speed,) . 

Separating out the individual fields of c->bFlags isn't necessary, Just print the whole of c->bFlags in a single snprintf(). That potentially removes the need for 32 different bit filters and snprintf's. 

I'm going to have to decline pull #37 for the same reason.

hhm0 commented:
But that's the thing. This pull doenst have any c code anymore; thats
what i was trying to explain in last post.

#37 on the other hand does, and i was not expecting you to pull that
one; i just put it there in case anyone is interested in applying it
to their local repo :-).

On 10/29/14, MalcolmRobb notifications@github.com wrote:

> The problem I have is the nature modification to the C code. Statements like
> ...
>   if (a->bFlags & MODES_ACFLAGS_SPEED_VALID) {
>       speed = 1;
>   }
> 
> followed later on by a snprintf( speed,) .
> 
> Separating out the individual fields of c->bFlags isn't necessary, Just
> print the whole of c->bFlags in a single snprintf(). That potentially
> removes the need for 32 different bit filters and snprintf's.
> 
> I'm going to have to decline pull #37 for the same reason.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/MalcolmRobb/dump1090/pull/18#issuecomment-60941467

hhm0 commented:
In other words the code in this pull no longer cantains any patch to
the c code; it only changes the web files in public_html/.

On 10/29/14, h hm heehooman@gmail.com wrote:

> But that's the thing. This pull doenst have any c code anymore; thats
> what i was trying to explain in last post.
> 
> #37 on the other hand does, and i was not expecting you to pull that
> one; i just put it there in case anyone is interested in applying it
> to their local repo :-).
> 
> On 10/29/14, MalcolmRobb notifications@github.com wrote:
> 
> > The problem I have is the nature modification to the C code. Statements
> > like
> > ...
> >   if (a->bFlags & MODES_ACFLAGS_SPEED_VALID) {
> >       speed = 1;
> >   }
> > 
> > followed later on by a snprintf( speed,) .
> > 
> > Separating out the individual fields of c->bFlags isn't necessary, Just
> > print the whole of c->bFlags in a single snprintf(). That potentially
> > removes the need for 32 different bit filters and snprintf's.
> > 
> > I'm going to have to decline pull #37 for the same reason.
> > 
> > ---
> > 
> > Reply to this email directly or view it on GitHub:
> > https://github.com/MalcolmRobb/dump1090/pull/18#issuecomment-60941467
